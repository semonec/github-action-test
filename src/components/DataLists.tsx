type Props = {
    users: Array<string>
}

const DataLists: React.FC<Props> = ({ users })=> {
    return (
        <div>
            {users.map(user => (<p key={user} className='single_user'>{user}</p>))}
        </div>
    )
}

export default DataLists