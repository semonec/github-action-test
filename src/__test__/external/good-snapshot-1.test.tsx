import { render, screen } from '@testing-library/react'

import DataLists from '../../components/DataLists'
import { createRenderer } from 'react-test-renderer/shallow'

describe('DataLists component', () => {
    // GOOD - test with manual assertions and snapshot to check for regressions
    it('renders single user without posts and with single_user class', () => {
        const users = ['user1']
        const renderer = createRenderer()

        render(<DataLists users={users} />)

        expect(screen.getByText('user1')).toBeInTheDocument()
        expect(screen.getByText('user1')).toHaveClass('single_user')

        renderer.render(<DataLists users={users} />)
        expect(renderer.getRenderOutput()).toMatchSnapshot()
    })
})