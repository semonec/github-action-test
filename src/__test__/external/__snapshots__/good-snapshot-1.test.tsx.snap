// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLists component renders single user without posts and with single_user class 1`] = `
<div>
  <p
    className="single_user"
  >
    user1
  </p>
</div>
`;
